### ISSUE TITLE
My setting page not working - Blank page 

### ISSUE THREAD

**Issue by anuj007a** [bitnami/bitnami-docker-testlink#116]
**Description**

```

My Settings page breaking ( Blank page ) due to /lib/usermanagement/userInfo.php HTTP/1.1" 500

```

**Steps to reproduce the issue:**
```
1. Login user valid credentials 
2. Click on My Settings option
```
**Describe the results you received:**
```
Blank page
```
**Describe the results you expected:**
```
Should open the current user setting page with edit permission like password reset 
```
**Additional information you deem important (e.g. issue happens only occasionally):**

```
**GET API is returning 500 http response:** /lib/usermanagement/userInfo.php HTTP/1.1" 500

**Server Exception:**


testlink    | [Sat Apr 24 18:09:02.828923 2021] [php7:error] [pid 180] [client 172.16.28.242:56360] PHP Fatal error:  Uncaught Error: Call to a member function hasRight() on null in /opt/bitnami/testlink/lib/functions/common.php:1925\nStack trace:\n#0 /opt/bitnami/testlink/lib/functions/common.php(1643): getGrantSetWithExit(Object(database), Object(stdClass), Object(testproject), Array)\n#1 /opt/bitnami/testlink/lib/usermanagement/userInfo.php(22): initUserEnv(Object(database), Object(stdClass))\n#2 {main}\n  thrown in /opt/bitnami/testlink/lib/functions/common.php on line 1925, referer: http://10.200.9.244/lib/general/navBar.php?tproject_id=0&tplan_id=0&updateMainPage=1
testlink    | 172.16.28.242 - - [24/Apr/2021:18:09:02 +0000] "GET /lib/usermanagement/userInfo.php HTTP/1.1" 500 -
testlink    | [Sat Apr 24 18:09:11.840867 2021] [php7:error] [pid 173] [client 172.16.28.242:56364] PHP Fatal error:  Uncaught Error: Call to a member function hasRight() on null in /opt/bitnami/testlink/lib/functions/common.php:1925\nStack trace:\n#0 /opt/bitnami/testlink/lib/functions/common.php(1643): getGrantSetWithExit(Object(database), Object(stdClass), Object(testproject), Array)\n#1 /opt/bitnami/testlink/lib/usermanagement/userInfo.php(22): initUserEnv(Object(database), Object(stdClass))\n#2 {main}\n  thrown in /opt/bitnami/testlink/lib/functions/common.php on line 1925, referer: http://10.200.9.244/lib/general/navBar.php?tproject_id=0&tplan_id=0&updateMainPage=1
testlink    | 172.16.28.242 - - [24/Apr/2021:18:09:11 +0000] "GET /lib/usermanagement/userInfo.php HTTP/1.1" 500 -
```

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:18:05 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:16:00 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

- Output of `docker info`:

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)

Server:
 Containers: 2
  Running: 2
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 20.10.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-1038-aws
 Operating System: Ubuntu 18.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.904GiB
 Name: ip-10-200-9-244
 ID: 75DB:PNJG:CQKS:MOX6:FNUH:HW5H:DXLS:Y2FO:TVIL:THK5:SXTM:YSV5
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support

```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.28.5, build unknown
docker-py version: 4.4.4
CPython version: 3.6.9
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic

```

**Comment by alemorcuq** [comment]
Hi, @anuj007a.

How are you deploying it? Using this docker compose file (https://raw.githubusercontent.com/bitnami/bitnami-docker-testlink/master/docker-compose.yml) I'm not able to reproduce the issue. After signing in with valid credentials, I can click the My Settings button and I don't receive a blank page:

```
testlink_1  | 172.23.0.1 - - [27/Apr/2021:09:50:57 +0000] "GET /lib/usermanagement/userInfo.php HTTP/1.1" 200 4775
```

![image](https://user-images.githubusercontent.com/47833905/116222807-33e78d00-a74f-11eb-9264-143b9213c66a.png)

**Comment by anuj007a** [comment]
Hi @alemorcuq,

Deployed using  docker compose file [https://github.com/bitnami/bitnami-docker-testlink/blob/master/docker-compose.yml](url)
 I am able to reproduce in my System. Please let me know if any information is required for debug the issue ?

**Comment by alemorcuq** [comment]
Can you share the full logs of your container?

**Comment by bdi-gud** [comment]
Hi,
I have the same issue.

It was after updating my project properties :
![image](https://user-images.githubusercontent.com/83651463/117009113-733c4d80-acfc-11eb-99e2-fa25f2102233.png)

Logs
```
IP - - [04/May/2021:13:15:55 +0000] "GET /lib/project/projectEdit.php?doAction=edit&tprojectID=1 HTTP/1.1" 200 3623


[Tue May 04 13:17:57.988453 2021] [php7:error] [pid 163] [client IP:65451] PHP Fatal error:  Uncaught Error: Call to a member function hasRight() on null in /opt/bitnami/testlink/lib/functions/common.php:1925\nStack trace:\n#0 /opt/bitnami/testlink/lib/functions/common.php(1643): getGrantSetWithExit(Object(database), Object(stdClass), Object(testproject), Array)\n#1 /opt/bitnami/testlink/lib/usermanagement/userInfo.php(22): initUserEnv(Object(database), Object(stdClass))\n#2 {main}\n  thrown in /opt/bitnami/testlink/lib/functions/common.php on line 1925, referer: http://IP:6001/lib/general/navBar.php?tproject_id=0&tplan_id=0&updateMainPage=1


IP - - [04/May/2021:13:17:57 +0000] "GET /lib/usermanagement/userInfo.php HTTP/1.1" 500 -
```

**Comment by alemorcuq** [comment]
Thank you for the information, @bdi-gud. I will try to enable those features myself to see if I can reproduce the issue.

**Comment by yx135790** [comment]
Same here , and this happens only when you checked the "Enable Inventory" of your first project. 
This PR mentioned about this.
https://github.com/TestLinkOpenSourceTRMS/testlink-code/pull/256
Also I check on the Testlink branch 2.0.0-20200601-01 , it has been fixed in this commit. FYI the default branch remains not fixed, I guess its not updated
https://github.com/TestLinkOpenSourceTRMS/testlink-code/commit/f5bc34dd613a821c9e143eb99a67b81d7b54caf2
OK, I think this is an known issue of 1.9.20 that are not fixed yet.

**Comment by migruiz4** [comment]
Hi @yx135790,

Thank you very much for sharing this information. So it appears to be an issue with Testlink and not the docker image, and it should be fixed with the next version of Testlink (looks like the fix will be included in version 2.0.0).

In that case, I will close this issue and we will wait for Testlink to release a new version.
